### PR TITLE
Split code to sources and headers

### PR DESCRIPTION
**Pull request by MatusLaza**
The code is split to sources and headers to allow writing tests.

### CODE REVIEW

File: sw_watchdog/include/sw_watchdog/windowed_watchdog.hpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+
+#include "std_msgs/msg/bool.hpp"
+#include "sw_watchdog_msgs/msg/heartbeat.hpp"
+#include "sw_watchdog_msgs/msg/status.hpp"
+#include "sw_watchdog/visibility_control.h"
+
+constexpr char OPTION_AUTO_START[] = "--activate";
```
Comment (nkoenig) [inline sw_watchdog/include/sw_watchdog/windowed_watchdog.hpp]: We don't need to expose these variables in the header file. It's safe to leave these in the `.cpp` file.

File: sw_watchdog/include/sw_watchdog/windowed_watchdog.hpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * Internally relies on the QoS deadline and liveliness policies provided by the rmw implementation
+ * (e.g., DDS). The lease passed to this watchdog has to be > the period of the heartbeat signal to
+ * account for network transmission times.
+ */
+class WindowedWatchdog : public rclcpp_lifecycle::LifecycleNode
+{
+public:
+  SW_WATCHDOG_PUBLIC
+  explicit WindowedWatchdog(const rclcpp::NodeOptions &);
+  void publish_status(uint16_t misses);
```
Comment (nkoenig) [inline sw_watchdog/include/sw_watchdog/windowed_watchdog.hpp:53]: Can you copy over the function documentation from the source files? For example, this function has the following doc string:
```
/// Publish lease expiry of the watched entity.
```

I think every function here has a one line documentation comment.

File: sw_watchdog/include/sw_watchdog/simple_watchdog.hpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+
+#include "std_msgs/msg/bool.hpp"
+#include "sw_watchdog_msgs/msg/heartbeat.hpp"
+#include "sw_watchdog_msgs/msg/status.hpp"
+#include "sw_watchdog/visibility_control.h"
+
+constexpr char OPTION_AUTO_START[] = "--activate";
```
Comment (nkoenig) [inline sw_watchdog/include/sw_watchdog/simple_watchdog.hpp]: You can leave these in the `.cpp` files. It'll reduce their exposure

File: sw_watchdog/include/sw_watchdog/simple_watchdog.hpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+public:
+  SW_WATCHDOG_PUBLIC
+  explicit SimpleWatchdog(const rclcpp::NodeOptions &);
+
+  /**
+   * @brief Publish watchdog status message
+   */
+  void publish_status();
+
+  NodeCallback on_configure(const rclcpp_lifecycle::State &);
```
Comment (nkoenig) [inline sw_watchdog/include/sw_watchdog/simple_watchdog.hpp:56]: Port over the function documentation please.

**Review by nkoenig — CHANGES_REQUESTED** [review]
Just some suggestions to reduce scope of a few variables and migrate documentation to the headers.

**Comment by MatusLaza** [comment]
All review comments were incorporated. Please review the code again.

**Review by nkoenig — APPROVED** [review]

**Pull request merged**